File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/SdJwtApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.sdjwt.SdJwtParser
import my.ssdid.sdk.domain.sdjwt.SdJwtVc
import my.ssdid.sdk.domain.sdjwt.StoredSdJwtVc
import my.ssdid.sdk.domain.vault.Vault

class SdJwtApi internal constructor(private val vault: Vault) {
    fun parse(compactSdJwt: String): SdJwtVc = SdJwtParser.parse(compactSdJwt)
    suspend fun store(sdJwtVc: StoredSdJwtVc): Result<Unit> = vault.storeStoredSdJwtVc(sdJwtVc)
    suspend fun list(): List<StoredSdJwtVc> = vault.listStoredSdJwtVcs()
}

File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/SdJwtApi.swift
import Foundation

/// Facade for SD-JWT VC operations (parse, store, list).
public struct SdJwtApi {
    let storage: VaultStorage

    public func parse(compactSdJwt: String) throws -> SdJwtVc {
        try SdJwtParser.parse(compactSdJwt)
    }

    public func store(_ sdJwtVc: StoredSdJwtVc) async throws {
        try await storage.saveSdJwtVc(sdJwtVc)
    }

    public func list() async -> [StoredSdJwtVc] {
        await storage.listSdJwtVcs()
    }
}
